### ISSUE TITLE
Write documentation for invenio-files-processor which contains info on how to use and how to extend it with more extractors.

### ISSUE THREAD

**Issue by krzysztof** [zenodo/zenodo#1274]

**Comment by krzysztof** [comment]
This task involves:
- [ ] Go through the code and supplement functions and classes docstrings with parameter types:
```
:param foobar: Foobar parameter
:type foobar: str
```
- [ ] Rendering Sphinx documentation and making sure that all pages are non-empty, all text renders nice and all API references have proper hyperlinks (if they do).

- [ ] Writing up usage guide which describes how to make use of pdfmetadata extractor
  - User needs to run grobid service - give an example on how to run it on docker:  ``docker run -t --rm -p 8080:8080 lfoppiano/grobid:0.4.1``
  - Describe the POST parameters and what they do, give some mockup responses which can be expected.
- [ ] Writing up usage guide which contains info on how to extent the extractor and how to use it.
Take a look at Inveino-Admin docs: https://invenio-admin.readthedocs.io/en/latest/usage.html#adding-admin-views-from-invenio-module for inspiration on how this guide looks like.
Basically you create an example of a new extractor and provide a short code snipplets and entry point registration example. Finally there should be some info on the REST API endpoint and how to use it.

**Comment by xiaom** [comment]
The doc is parts of https://github.com/inveniosoftware/invenio-files-processor/pull/9
